src/odemis/driver/test/focustracker_test.py

```diff
@@ -26,162 +26,82 @@
-
-    def test_target_pos(self):
-        """Verify that the target position can be read and not written."""
-        self.assertAlmostEqual(self.fake_focus_tracker.target_pos.value, 2e-5)
-
-    def test_switch_tracking(self):
-        """Verify that the focus tracker can switch between tracking and untracking."""
-        self.fake_focus_tracker.tracking.value = True
-        self.assertTrue(self.fake_focus_tracker.tracking.value)
-        self.assertTrue(self.fake_focus_tracker.get_tracking())
-        self.fake_focus_tracker.tracking.value = False
-        self.assertFalse(self.fake_focus_tracker.tracking.value)
-        self.assertFalse(self.fake_focus_tracker.get_tracking())
-
-    def test_pid_gains(self):
-        """Test that PID gains are set when within range and remain unchanged when new gain is out of range."""
-        new_p = 0.5
-        new_i = 15
-        new_d = 3
-        self.fake_focus_tracker.updateMetadata({model.MD_GAIN_P: new_p, model.MD_GAIN_I: new_i, model.MD_GAIN_D: new_d})
-        set_p = self.fake_focus_tracker.get_proportional()
-        self.assertEqual(new_p, set_p)
-        set_i = self.fake_focus_tracker.get_integral()
-        self.assertEqual(new_i, set_i)
-        set_d = self.fake_focus_tracker.get_derivative()
-        self.assertEqual(new_d, set_d)
+if TEST_NOHW:
+    KWARGS = {"name": "Focus Tracker", "role": "focus-tracker", "node_idx": 0x10, "channel": 'fake'}
+else:
+    KWARGS = {"name": "Focus Tracker", "role": "focus-tracker", "node_idx": 0x10, "channel": 'can0'}
 
 
 class TestFocusTrackerCO(unittest.TestCase):
     """Test the focus tracker functionality with the CANopen network."""
-    backend_was_running = False
-
-    @classmethod
-    def setUpClass(cls):
-        try:
-            test.start_backend(FOCUS_TRACKER_CONFIG)
-        except LookupError:
-            logging.info("A running backend is already found, skipping tests")
-            cls.backend_was_running = True
-            return
-        except IOError as exp:
-            logging.error(str(exp))
-            raise
-
-        # find components by their role
-        cls.focus_tracker = model.getComponent(role="focus-tracker")
-
-    @classmethod
-    def tearDownClass(cls):
-        if cls.backend_was_running:
-            return
-        test.stop_backend()
+
+    def setUp(self):
+        self.focus_tracker = FocusTrackerCO(**KWARGS)
+        self.kwargs = KWARGS
 
     def test_connection(self):
         """Test that connection to the network is successful when connected, and unsuccessful when disconnected."""
-        # If the network scanner does not see a node at index 0x10, the focus tracker is not connected.
-        self.assertTrue(0x10 in self.focus_tracker.get_available_nodes())
+        if TEST_NOHW:
+            # If the network scanner does not see a node at index 0x10, the focus tracker is not connected.
+            self.assertFalse(0x10 in self.focus_tracker.network.scanner.nodes)
```

Comment:
I don't think it's a really good proof that the code is broken if the simulator reports that 0x10 is present. I'd suggest to just not test (so the simulator may or may not report the presence of 0x10).